python-checks/src/main/java/org/sonar/python/checks/tests/DedicatedAssertionCheck.java

```diff
@@ -0,0 +1,239 @@
+    Map<String, String> suggestions = isAssertTrue ? ASSERT_TRUE_SUGGESTIONS : ASSERT_FALSE_SUGGESTIONS;
+    List<Argument> arguments = callExpression.arguments();
+    if (arguments.isEmpty()) {
+      return;
+    }
+    RegularArgument firstArg = TreeUtils.nthArgumentOrKeyword(0, "expr", arguments);
+    if (firstArg == null) {
+      return;
+    }
+    Expression expression = Expressions.removeParentheses(firstArg.expression());
+    String dedicatedAssertion = null;
+    if (expression.is(Tree.Kind.COMPARISON)) {
+      BinaryExpression binaryExpression = (BinaryExpression) expression;
+      if (binaryExpression.leftOperand().is(Tree.Kind.COMPARISON)) {
+        // Avoid reporting on chained comparisons (due to how comparisons are defined in the grammar, only the left operand can be a comparison)
+        return;
+      }
+      dedicatedAssertion = suggestions.get(binaryExpression.operator().value());
+    } else if (expression.is(Tree.Kind.IN)) {
+      dedicatedAssertion = suggestions.get(IN);
+    } else if (expression.is(Tree.Kind.IS)) {
+      dedicatedAssertion = ((IsExpression) expression).notToken() == null ? suggestions.get(IS) : suggestions.get(IS_NOT);
+    } else if (isCallTo(IS_INSTANCE, expression)) {
+      dedicatedAssertion = suggestions.get(IS_INSTANCE);
+    }
+    if (dedicatedAssertion != null) {
+      ctx.addIssue(callExpression, String.format(MESSAGE, dedicatedAssertion));
+    }
+  }
+
+  private static void checkAssertEqualOrNotEqual(CallExpression callExpression, SubscriptionContext ctx, boolean isAssertEqual) {
+    List<Argument> arguments = callExpression.arguments();
+    if (arguments.size() < 2) {
+      return;
+    }
+    RegularArgument firstArg = TreeUtils.nthArgumentOrKeyword(0, FIRST, arguments);
+    RegularArgument secondArg = TreeUtils.nthArgumentOrKeyword(1, SECOND, arguments);
+    if (firstArg == null || secondArg == null) {
+      return;
+    }
+
+    Expression firstExpression = Expressions.removeParentheses(firstArg.expression());
+    Expression secondExpression = Expressions.removeParentheses(secondArg.expression());
+    String firstDedicatedAssertion = dedicatedAssertion(firstExpression, isAssertEqual);
+    if (firstDedicatedAssertion != null) {
+      // If two suggestions are possible, priority is given to first argument, as it is conventionally the "expected" value
+      ctx.addIssue(callExpression, String.format(MESSAGE, firstDedicatedAssertion));
+      return;
+    }
+    String secondDedicatedAssertion = dedicatedAssertion(secondExpression, isAssertEqual);
+    if (secondDedicatedAssertion != null) {
+      ctx.addIssue(callExpression, String.format(MESSAGE, secondDedicatedAssertion));
+    }
+  }
+
+  private static void checkAssertAlmostEqualOrNot(CallExpression callExpression, SubscriptionContext ctx, String methodName) {
+    List<Argument> arguments = callExpression.arguments();
+    if (arguments.size() < 2) {
+      return;
+    }
```

Comment:
Same as above, maybe this check can be removed as this use case is handled by the next check.